### PR TITLE
CoredumpFilter: fix stack overflow and invalid assignment with 'all'

### PR DESCRIPTION
**Pull request by bluca**

### CODE REVIEW

File: src/shared/coredump-util.h [hunk 1]
```diff
@@ -22,6 +22,9 @@ typedef enum CoredumpFilter {
                                       1u << COREDUMP_FILTER_ELF_HEADERS | \
                                       1u << COREDUMP_FILTER_PRIVATE_HUGE)
 
+/* The kernel doesn't like UINT_MAX and returns ERANGE, use UINT32_MAX to support future new flags */
```
Comment (daandemeyer) [inline src/shared/coredump-util.h]: UINT_MAX => UINT64_MAX?

**Review by DaanDeMeyer — APPROVED** [review]

**Review by yuwata — APPROVED** [review]

File: src/shared/coredump-util.c [hunk 2]
```diff
@@ -158,7 +158,7 @@ int parse_auxv(int log_level,
 }
 
 int set_coredump_filter(uint64_t value) {
-        char t[STRLEN("0xFFFFFFFF")];
+        char t[DECIMAL_STR_MAX(uint64_t)]; /* For the same number decimal >> hex */
```
Comment (poettering) [inline src/shared/coredump-util.c]: hmm, i find this icky. if you did the same for uint8_t, then DECIMAL_STR_MAX(uint8_t) would return 4 (i.e. `sizeof("255")`). But actually you need 5 bytes space, i.e. `sizeof("0xFF")`...

Hence, maybe just do it right:

```c
/* Maximum buffer size needed for formatting an unsigned integer type as hex, including space for `0x` prefix and trailing NUL suffix */
#define HEXADECIMAL_STR_MAX(type) (2+sizeof(type)*2+1)
```

It's short, then requires no further comment, and is correct for any unsigned integer type. yay!

File: src/shared/coredump-util.c [hunk 3]
```diff
@@ -158,7 +158,7 @@ int parse_auxv(int log_level,
...
 
         sprintf(t, "0x%"PRIx64, value);
```
Comment (poettering) [inline src/shared/coredump-util.c]: this should use xsprintf() btw

File: src/shared/coredump-util.h [hunk 4]
```diff
@@ -22,6 +22,9 @@ typedef enum CoredumpFilter {
                                       1u << COREDUMP_FILTER_ELF_HEADERS | \
                                       1u << COREDUMP_FILTER_PRIVATE_HUGE)
```
Comment (poettering) [inline src/shared/coredump-util.h:23]: maybe change `1u` → `UINT32_C(1)` at the same time?

Comment (bluca) [reply src/shared/coredump-util.h:23]: let's leave that for another day, doesn't make a difference anyway

**Comment by keszybz** [comment]
LGTM too.

File: src/shared/coredump-util.h (hunk 1 above)
Comment (yuwata) [reply src/shared/coredump-util.h]: Not addressed??

File: src/shared/coredump-util.h [hunk 5]
```diff
@@ -22,7 +22,7 @@ typedef enum CoredumpFilter {
...
 
-/* The kernel doesn't like UINT_MAX and returns ERANGE, use UINT32_MAX to support future new flags */
+/* The kernel doesn't like UINT64_MAX and returns ERANGE, use UINT32_MAX to support future new flags */
```
Comment (yuwata) [inline src/shared/coredump-util.h:25]: Ah, the change is mixed... Please move the change into the second commit.

**Comment by poettering** [comment]
lgtm

**Pull request merged**